server/src/commands/codeaction.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import { CodeAction, CodeActionKind, CodeActionParams, Command, TextDocumentEdit, TextEdit, WorkspaceEdit } from 'vscode-languageserver';
+import { log } from '../state';
+import { LanguageServerState } from '../types';
+import * as semver from 'semver';
+import { splitLines } from '../utils';
+
+function createIncrementNamespaceVersionWorkspaceEdit(state: LanguageServerState, params: CodeActionParams, namespace: string, oldVersion: string, newVersion: string): WorkspaceEdit {
+	const textEdit: TextEdit = {
+		range: {
+			start: params.range.start,
+			end: params.range.end
+		},
+		newText: `namespace ${namespace}@${newVersion}\n`
+	};
+
+	// update the namespace in the current cto
+	const textDocumentEdit = TextDocumentEdit.create({
+		uri: params.textDocument.uri,
+		version: 1
+	}, [textEdit]);
+	const textDocumentEdits: TextDocumentEdit[] = [];
+	textDocumentEdits.push(textDocumentEdit);
+
+	// find all the documents that import this namespace and update their imports
+	// import <old versioned ns>.{Foo}
+	// import <old versioned ns>.Foo
```

Comment:
this should work fine with `import <old versioned ns>.{Foo as aliasedFoo}` too, right?

Reply:
Yes, I checked and it works. Thanks for the heads up though, I enabled the import aliasing...